src/ansys/fluent/core/solver/flobject.py

```diff
@@ -363,6 +378,27 @@ class Real(SettingsBase[RealType], Numerical):
     expression values.
     """
 
+    def get_state_as_quantity(self) -> Optional[Quantity]:
+        """Get the state of the object as a Quantity."""
+        try:
+            unit = self._quantity_map(self.get_attr("units-quantity"))
+            value = self.get_state()
+            quantity = Quantity(value, quantity_table={unit: 1})
+            return quantity
+        except (ValueError, KeyError, ModuleNotFoundError) as e:
+            warnings.warn(f"Unable to construct 'Quantity'. {e}")
+
+    def set_state(self, state: Optional[StateT] = None, **kwargs):
+        """Set the state of the object."""
+        if isinstance(state, Quantity):
+            unit = self._quantity_map(self.get_attr("units-quantity"))
+            state = state.to(Unit(table={unit: 1}))
+            return self.flproxy.set_var(
+                self.path, self.to_scheme_keys(state.si_value), **kwargs
```

Comment:
@seanpearsonuk @Adam-Boutin `si_value` is not available on the Quantity instance. Therefore we get assertion error due to incorrect conversion as follows.

![image](https://github.com/ansys/pyfluent/assets/106588300/3f80165e-9fb0-4104-8306-6b3965d68d3d)

I have updated it as follows - 

![image](https://github.com/ansys/pyfluent/assets/106588300/34623c2a-3e02-4110-a2f0-e1787799457b)

Now this error is resolved now.